bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/LGThinQHandlerFactory.java

```diff
@@ -0,0 +1,136 @@
+import static org.openhab.binding.lgthinq.internal.LGThinQBindingConstants.THING_TYPE_DISHWASHER;
+import static org.openhab.binding.lgthinq.internal.LGThinQBindingConstants.THING_TYPE_DRYER;
+import static org.openhab.binding.lgthinq.internal.LGThinQBindingConstants.THING_TYPE_DRYER_TOWER;
+import static org.openhab.binding.lgthinq.internal.LGThinQBindingConstants.THING_TYPE_FRIDGE;
+import static org.openhab.binding.lgthinq.internal.LGThinQBindingConstants.THING_TYPE_HEAT_PUMP;
+import static org.openhab.binding.lgthinq.internal.LGThinQBindingConstants.THING_TYPE_WASHING_MACHINE;
+import static org.openhab.binding.lgthinq.internal.LGThinQBindingConstants.THING_TYPE_WASHING_TOWER;
+
+import java.util.Objects;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.lgthinq.internal.handler.LGThinQAirConditionerHandler;
+import org.openhab.binding.lgthinq.internal.handler.LGThinQBridgeHandler;
+import org.openhab.binding.lgthinq.internal.handler.LGThinQDishWasherHandler;
+import org.openhab.binding.lgthinq.internal.handler.LGThinQFridgeHandler;
+import org.openhab.binding.lgthinq.internal.handler.LGThinQWasherDryerHandler;
+import org.openhab.binding.lgthinq.internal.type.ThinqChannelGroupTypeProvider;
+import org.openhab.binding.lgthinq.internal.type.ThinqChannelTypeProvider;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.openhab.core.thing.link.ItemChannelLinkRegistry;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link LGThinQHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Nemer Daud - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = ThingHandlerFactory.class, configurationPid = "binding.lgthinq")
+public class LGThinQHandlerFactory extends BaseThingHandlerFactory {
+
+    private final Logger logger = LoggerFactory.getLogger(LGThinQHandlerFactory.class);
+
+    private final HttpClientFactory httpClientFactory;
+
+    private final LGThinQStateDescriptionProvider stateDescriptionProvider;
+
+    @Nullable
+    @Reference
+    protected ThinqChannelTypeProvider thinqChannelProvider;
+    @Nullable
+    @Reference
+    protected ThinqChannelGroupTypeProvider thinqChannelGroupProvider;
+    @Nullable
+    @Reference
+    protected ItemChannelLinkRegistry itemChannelLinkRegistry;
```

Comment:
Why is this not added to the constructor just like the `HttpClientFactory` and other `@Reference` ? Seems more consistent and makes all the `Objects.requireNonNull` obsolete.